### PR TITLE
Signup and Login

### PR DESCRIPTION
**Pull request by swashcap**
Asana tasks:

* Login: https://app.asana.com/0/37131652076106/46141439482573
* Registration: https://app.asana.com/0/37131652076106/46141439482548

### Goal

Enable user login, logout and new user sign up.

### Testing

1. Start up Coinstac and remove the auth local storage items:

  ```js
  localStorage.removeItem('COINSTAC_AUTH_RESPONSE');
  localStorage.removeItem('COINSTAC_USER');
  ```

2. Refresh the page to get accurate application state
3. Try to log in. Your username and email should be in the bottom left corner of the app.
4. Try to sign up. The form should catch basic errors, and it should redirect to the “Home” route on success.
5. Try to log out. It should redirect you to the login form.

### CODE REVIEW

File: app/js/components/form-login-controller.js [hunk 1]
```diff
@@ -9,39 +9,18 @@ import User from '../models/user';
...
-            name: 'Admin McAdmin',
-            institution: 'BillBraskeyLTD',
-            email: 'heyo',
-        });
-
-        app.notifications.push({
-            message: 'Welcome ' + user.name,
-            level: 'success'
-        });
-        app.router.transitionTo('/home');
```
Comment (cdaringe) [inline app/js/components/form-login-controller.js]: noooooooo :+1:

File: app/js/components/form-login-controller.js [hunk 2]
```diff
@@ -9,39 +9,18 @@ import User from '../models/user';
...
 
         auth.login(this.refs.logon.data())
             .then(user => {
                 app.notifications.push({
-                    message: 'Welcome ' + user.name,
+                    message: `Welcome, ${user.label}!`,
                     level: 'success'
                 });
-                app.router.transitionTo('/home');
+                app.router.transitionTo('home');
```
Comment (cdaringe) [inline app/js/components/form-login-controller.js]: really, no more `'/'`?

File: app/js/services/auth.js [hunk 3]
```diff
@@ -99,48 +114,53 @@ const Auth = {
...
+    setUser: function(user) {
+        localStorage.setItem(USER_KEY, JSON.stringify(user));
         return Auth.getUser();
     },
 
     /**
      * Get saved user.
      *
-     * @return {Object|undefined}
+     * @return {(Object|undefined)}
```
Comment (cdaringe) [inline app/js/services/auth.js]: huh, `()` is the way. who knew. jsdoc confirmed!

**Comment by cdaringe** [comment]
this is rad! super works.

screen shot 2015-10-12 at 6 38 22 pm

There was an issue with logout, but I'm going to merge anyway

File: app/js/components/form-login-controller.js (hunk 2 above)
Comment (swashcap) [inline app/js/components/form-login-controller.js]: This change may not be necessary. I might have been trying to fix the automatic login. :frowning:

**Pull request merged**